tensorflow/lite/delegates/nnapi/java/src/main/native/nnapi_delegate_jni.cc

```diff
@@ -21,10 +21,28 @@ limitations under the License.
 extern "C" {
 #endif  // __cplusplus
 
+using namespace tflite;
+
 JNIEXPORT jlong JNICALL
-Java_org_tensorflow_lite_nnapi_NnApiDelegate_createDelegate(JNIEnv* env,
-                                                            jclass clazz) {
-  return reinterpret_cast<jlong>(tflite::NnApiDelegate());
+Java_org_tensorflow_lite_nnapi_NnApiDelegate_createDelegate(
+    JNIEnv* env, jclass clazz, jint preference, jstring accelerator_name,
+    jstring cache_dir, jstring model_token) {
+
+  StatefulNnApiDelegate::Options options = StatefulNnApiDelegate::Options();
+  options.execution_preference =
+      (StatefulNnApiDelegate::Options::ExecutionPreference)preference;
+  options.accelerator_name = env->GetStringUTFChars(accelerator_name, NULL);
```

Comment:
You need to make null checks here for the arguments, e.g.,

if (accelerator_name) {
  options.accelerator_name = env->GetStringUTFChars(accelerator_name, NULL);
}
...

Reply:
Thanks. Did that, please check the latest commit.

Reply:
Thanks. Did that, please check the latest commit.

Reply:
Thanks. Did that, please check the latest commit.